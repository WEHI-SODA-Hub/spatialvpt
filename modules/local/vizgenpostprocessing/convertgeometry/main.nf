process VIZGENPOSTPROCESSING_CONVERTGEOMETRY {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container 'nf-core/vizgen-postprocessing_container:v0.1.1'

    input:
    tuple val(meta), path(boundary_dir), val(boundary_regex)

    output:
    tuple val(meta), path("*.parquet"), emit: segmented_tile
    path  "versions.yml"              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    vpt --verbose \\
        convert-geometry \\
        $args \\
        --input-boundaries ${boundary_dir}/${boundary_regex} \\
        --output-boundaries ${prefix}.parquet

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vpt: \$( pip show vpt | grep Version | sed -e "s/Version: //g" )
        vpt-plugin-cellpose2: \$( pip show vpt-plugin-cellpose2 | grep Version | sed -e "s/Version: //g" )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.parquet

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vpt: \$( pip show vpt | grep Version | sed -e "s/Version: //g" )
        vpt-plugin-cellpose2: \$( pip show vpt-plugin-cellpose2 | grep Version | sed -e "s/Version: //g" )
    END_VERSIONS
    """
}
