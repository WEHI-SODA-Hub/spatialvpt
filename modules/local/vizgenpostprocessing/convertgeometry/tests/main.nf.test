nextflow_process {

    name "Test Process VIZGENPOSTPROCESSING_CONVERTGEOMETRY"
    script "../main.nf"
    process "VIZGENPOSTPROCESSING_CONVERTGEOMETRY"

    tag "modules"
    tag "modules_nfcore"
    tag "vizgenpostprocessing"
    tag "vizgenpostprocessing/convertgeometry"

    test("vizgen - boundary_dir") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test_sample' ], // meta map
                    file("${projectDir}/tests/data", checkIfExists: true), // boundary_dir
                    '*.geojson' // boundary_regex
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }

    }

    test("vizgen - boundary_dir - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test_sample' ], // meta map
                    file("${projectDir}/tests/data", checkIfExists: true), // boundary_dir
                    '*.geojson' // boundary_regex
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }

    }

}
