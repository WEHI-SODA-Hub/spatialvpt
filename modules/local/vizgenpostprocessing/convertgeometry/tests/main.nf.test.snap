{
    "vizgen - boundary_dir": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.parquet:md5,9396c92e0a0b584ac7c4e7c7693e50e8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4ae16511545413e77168c557a19306a8"
                ],
                "segmentation": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.parquet:md5,9396c92e0a0b584ac7c4e7c7693e50e8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4ae16511545413e77168c557a19306a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-08T16:36:56.684331006"
    },
    "vizgen - boundary_dir - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.parquet:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4ae16511545413e77168c557a19306a8"
                ],
                "segmentation": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.parquet:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4ae16511545413e77168c557a19306a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-08T16:34:34.093052973"
    }
}